client/tmclient/client.py

```diff
@@ -0,0 +1,225 @@
+import os, time
+import asyncio
+import json
+import websockets
+import urwid
+
+from .config import Config
+from . import ui
+from .ui import Screen, Form, FormField, menu, menu_button, sub_menu
+
+LOOP = asyncio.get_event_loop()
+CLIENT = None
+
+
+def quit(_):
+    raise urwid.ExitMainLoop()
+
+class Client:
+    def __init__(self):
+        global CLIENT
+        CLIENT = self
+        self.connection = None
+        self.config = Config()
+        self.ui = ui.UI(LOOP)
+        self.listening = False
+        self.authenticated = False
+        self.ui.base = urwid.Overlay(
+            urwid.Filler(urwid.Text('connecting..', align='center')),
+            ui.solidfill('░', 'background'),
+            align='center', width=15,
+            valign='middle', height=3,)
+        asyncio.wait_for(asyncio.ensure_future(self.connect(), loop=LOOP),60.0, loop=LOOP)
+        self.ui.loop.run()
```

Comment:
i'd prefer the run to be explicit and not an implicit side effect of creating a client.